ValveResourceFormat/Resource/ResourceTypes/Morph.cs

```diff
@@ -0,0 +1,201 @@
+                    var rectWidth = (int)Math.Round(rect.GetFloatProperty("m_flUWidthSrc") * texWidth, 0);
+                    var rectHeight = (int)Math.Round(rect.GetFloatProperty("m_flVHeightSrc") * texHeight, 0);
+                    var bundleDatas = rect.GetSubCollection("m_bundleDatas");
+                    foreach (var bundleData in bundleDatas)
+                    {
+                        var bundleType = int.Parse(bundleData.Key);
+                        var bundle = bundleData.Value as IKeyValueCollection;
+                        var rectU = (int)Math.Round(bundle.GetFloatProperty("m_flULeftSrc") * texWidth, 0);
+                        var rectV = (int)Math.Round(bundle.GetFloatProperty("m_flVTopSrc") * texHeight, 0);
+                        var ranges = bundle.GetFloatArray("m_ranges");
+                        var offsets = bundle.GetFloatArray("m_offsets");
+
+                        for (var row = rectV; row < rectV + rectHeight; row++)
+                        {
+                            for (var col = rectU; col < rectU + rectWidth; col++)
+                            {
+                                var colorIndex = row * texWidth + col;
+                                var color = texPixels[colorIndex];
+                                var dstI = row - rectV + yTopDst;
+                                var dstJ = col - rectU + xLeftDst;
+                                rectData[bundleType, dstI, dstJ] = new Vector3(
+                                    color.Red / 255f * ranges[0] + offsets[0],
+                                    color.Green / 255f * ranges[1] + offsets[1],
+                                    color.Blue / 255f * ranges[2] + offsets[2]
+                                    );
+                            }
+                        }
+                    }
+                }
+
+                var flexDataWithType = new Dictionary<int, List<byte>>();
+                for (int c = 0; c < bundleTypeCount; c++)
+                {
+                    var tmp = new List<byte>(width * height * BytesPerVertex);
+                    for (int i = 0; i < height; i++)
+                    {
+                        for (int j = 0; j < width; j++)
+                        {
+                            tmp.AddRange(BitConverter.GetBytes(rectData[c,i,j].X));
+                            tmp.AddRange(BitConverter.GetBytes(rectData[c,i,j].Y));
+                            tmp.AddRange(BitConverter.GetBytes(rectData[c,i,j].Z));
+                        }
+                    }
+                    flexDataWithType.Add(c, tmp);
+                }
+                _flexData.Add(morphName, flexDataWithType);
+            }
+        }
+
+        public bool CheckBundleId(int bundleId)
+        {
+            var bundleEnumName = _bundleTypes.GetStringProperty(bundleId.ToString());
+            if (bundleEnumName == "MORPH_BUNDLE_TYPE_POSITION_SPEED" || bundleEnumName == "BUNDLE_TYPE_POSITION_SPEED")
```

Comment:
Is the enum value `1` in both cases?

```
enum MorphBundleType
{
  MORPH_BUNDLE_TYPE_NONE = 0,
  MORPH_BUNDLE_TYPE_POSITION_SPEED = 1,
  MORPH_BUNDLE_TYPE_NORMAL_WRINKLE = 2,
}
```

Reply:
Yes, but MORPH_BUNDLE_TYPE_POSITION_SPEED is a string and BUNDLE_TYPE_POSITION_SPEED is a enum.

Reply:
Ah yes, the classic.

See this for example, you can do the same:

https://github.com/SteamDatabase/ValveResourceFormat/blob/33ba0aa5431f108376f5413e6d95aa3f9e085ed1/GUI/Types/Renderer/RenderableMesh.cs#L181-L196